android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -1509,78 +1515,57 @@ public static void getTxExtraInfo(BraveWalletBaseActivity activity, NetworkInfo
                 BraveWalletConstants.P3A_COUNT_TEST_NETWORKS_SWITCH);
-
-        keyringService.getKeyringInfo(getKeyringForCoinType(coinType), keyringInfo -> {
-            jsonRpcService.getAllNetworks(coinType, networks -> {
-                ArrayList<NetworkInfo> relevantNetworks = new ArrayList<NetworkInfo>();
-                for (NetworkInfo net : networks) {
-                    // Exclude testnet chain data by default.
-                    // Testnet chain counting can be enabled via the
-                    // --p3a-count-wallet-test-networks switch
-                    if (countTestNetworks
-                            || !WalletConstants.KNOWN_TEST_CHAIN_IDS.contains(net.chainId)) {
-                        relevantNetworks.add(net);
-                    }
-                }
-                AsyncUtils.MultiResponseHandler multiResponse =
-                        new AsyncUtils.MultiResponseHandler(relevantNetworks.size());
-                ArrayList<AsyncUtils.GetTxExtraInfoResponseContext> txExtraInfoResponses =
-                        new ArrayList<AsyncUtils.GetTxExtraInfoResponseContext>();
-                for (NetworkInfo net : relevantNetworks) {
-                    AsyncUtils.GetTxExtraInfoResponseContext ctx =
-                            new AsyncUtils.GetTxExtraInfoResponseContext(
-                                    multiResponse.singleResponseComplete);
-                    getTxExtraInfo(activity, net, keyringInfo.accountInfos, null, true, ctx);
-                    txExtraInfoResponses.add(ctx);
-                }
-                multiResponse.setWhenAllCompletedAction(() -> {
-                    HashSet<String> activeAddresses = new HashSet();
-                    for (AsyncUtils.GetTxExtraInfoResponseContext ctx : txExtraInfoResponses) {
-                        // If account has a non-zero native coin balance or blockchain token
-                        // balance, add the account address to the set.
-                        for (Map.Entry<String, Double> entry :
-                                ctx.nativeAssetsBalances.entrySet()) {
-                            if (entry.getValue() > 0.0d) {
-                                activeAddresses.add(entry.getKey());
-                            }
-                        }
-                        for (Map.Entry<String, HashMap<String, Double>> accEntry :
-                                ctx.blockchainTokensBalances.entrySet()) {
-                            for (Map.Entry<String, Double> tokenEntry :
-                                    accEntry.getValue().entrySet()) {
-                                if (tokenEntry.getValue() > 0.0d) {
-                                    activeAddresses.add(accEntry.getKey());
-                                }
-                            }
+        LiveDataUtil.observeOnce(
+                activity.getWalletModel().getCryptoModel().getNetworkModel().mCryptoNetworks,
+                allNetworks -> {
```

Comment:
nit: it's okay to use this for now though always prefer passing the values to static util methods directly as they can be called anytime, not lifecycle aware and often not programmed to handle unexpected scenarios.

Reply:
Ok I see. I'll make a fix later.